internal/osutils/autostart/autostart_mac.go

```diff
@@ -30,18 +41,26 @@ func (a *App) enable() error {
 		return errs.Wrap(err, "Could not get launch file")
 	}
 
-	launchFile, err := assets.ReadFileBytes(launchFileMacOS)
+	asset, err := assets.ReadFileBytes(launchFileSource)
 	if err != nil {
 		return errs.Wrap(err, "Could not read asset")
 	}
-	err = fileutils.WriteFile(path, launchFile)
+
+	content, err := strutils.ParseTemplate(
+		string(asset),
+		map[string]interface{}{"Exec": a.Exec, "Args": strings.Join(a.Args, " ")})
+	if err != nil {
+		return errs.Wrap(err, "Could not parse %s", a.options.launchFileName)
+	}
+
+	err = fileutils.WriteFile(path, []byte(content))
 	if err != nil {
```

Comment:
```suggestion
	if err = fileutils.WriteFile(path, []byte(content)); err != nil {
```

Try to avoid reusing err variables. It can give issues asserting nil values if the types satisfying the error interface don't match.

Reply:
What exactly is giving issues with this pattern? Is it the language itself or something else? If a package like our `errs` package is forcing us into a certain pattern then we should file a story to determine why.